merlin-worker/src/main/java/gov/nasa/jpl/aerie/merlin/worker/MerlinWorkerAppDriver.java

```diff
@@ -84,7 +86,9 @@ public static void main(String[] args) throws InterruptedException {
             notification.simulationTemplateRevision());
         final ResultsProtocol.WriterRole writer = owner.get();
         try {
-          simulationAgent.simulate(planId, revisionData, writer);
+          canceledListener.register(new DatasetId(datasetId));
+          simulationAgent.simulate(planId, revisionData, writer, canceledListener);
+          canceledListener.unregister();
         } catch (final Throwable ex) {
```

Comment:
🔵 I suspect that we might be able to do without `unregister` 🤔 

If we do keep `unregister`, we should consider what happens in the case that an exception is thrown during simulation. Perhaps moving it into a `finally` block, or making use of try-with-resources would ensure that it will definitely get called.

Reply:
I've moved the `unregister` to a `finally` block 👍 

Re: using a `try-with-resources`: I did initially try to make the listener implement `Autoclosable`, however I didn't think it matched the interface's typical pattern (that being that `close` is called at the end of the object's lifecycle). I believe I also thought that `try-with-resources` had to create the object in the "with resources" part, but apparently that was just a misconception on my end.

Re: not having `unregister`: In my opinion, by having `unregister` and by calling it immediately after `simulate`, we're guaranteeing that once the worker is done with a request, it is done paying attention to that any information about that request. I felt that this was logically cleaner and less likely to cause potential future bugs then having the worker continue to acknowledge canceled requests in the time between it completing a request and it claiming a new one.

I'm happy to discuss either of these further, as I can see the merit of both methods (though obviously not at the same time).